lib/sync.js

```diff
@@ -36,7 +36,25 @@ var DEBUG = true;
 // beyond that and wedge forever, so we need to track how long we are willing
 // to keep open the connection. This constant is *ADDED* to the timeout= value
 // to determine the max time we're willing to wait.
-var BUFFER_PERIOD_MS = 80 * 1000;
+var BUFFER_PERIOD_MS = 5 * 1000;
```

Comment:
5s seems a bit low here.  Is there a risk that the request an initial sync with a small (zero?) timeout - and then give only 5s for all the data to actually xfer down the pipe?